packages/report-flaky-tests/lib/github-api.ts

```diff
@@ -0,0 +1,89 @@
+/**
+ * External dependencies
+ */
+import * as github from '@actions/github';
+import * as unzipper from 'unzipper';
+
+/**
+ * Internal dependencies
+ */
+import type { FlakyTestResult } from './types';
+
+class GitHubAPI {
+	octokit: ReturnType< typeof github.getOctokit >;
+
+	constructor( token: string ) {
+		this.octokit = github.getOctokit( token );
+	}
+
+	get headBranch() {
+		return github.context.payload.workflow_run.head_branch;
+	}
+
+	get runURL() {
+		return github.context.payload.workflow_run.html_url;
+	}
+
+	async downloadReportFromArtifact(
+		artifactNamePrefix: string
+	): Promise< FlakyTestResult[] | undefined > {
+		const {
+			data: { artifacts },
+		} = await this.octokit.rest.actions.listWorkflowRunArtifacts( {
+			...github.context.repo,
+			run_id: github.context.payload.workflow_run.id,
+		} );
+
+		const matchArtifact = artifacts.find( ( artifact ) =>
+			artifact.name.startsWith( artifactNamePrefix )
+		);
+
+		if ( ! matchArtifact ) {
+			return undefined;
+		}
+
+		const download = await this.octokit.rest.actions.downloadArtifact( {
+			...github.context.repo,
+			artifact_id: matchArtifact.id,
+			archive_format: 'zip',
+		} );
+
+		const { files } = await unzipper.Open.buffer(
+			Buffer.from( download.data as Buffer )
+		);
+		const fileBuffers = await Promise.all(
+			files.map( ( file ) => file.buffer() )
+		);
+		const parsedFiles = fileBuffers.map(
+			( buffer ) => JSON.parse( buffer.toString() ) as FlakyTestResult
```

Comment:
What if it's invalid JSON? Or is missing fields that have recently been added to `FlakyTestResult`?

Reply:
> What if it's invalid JSON?

Then I believe the whole action would just fail. I don't think we can do anything about that? Just to make sure the artifact being uploaded has valid JSON.

> Or is missing fields that have recently been added to `FlakyTestResult`?

Good question! Not sure about that though. The action will always run on `trunk` so I don't think we have to worry about new fields? But there might be a problem with outdated artifacts though. I guess we just have to make sure it's backward compatible, or introduce something like `version` field. Open to ideas!

Reply:
A `version` field could be good. If nothing else it might encourage future developers to consider this problem before adding or changing a field.

Reply:
Added a `version` field to the reports!